Establish atix main

sbernhard opened the pull request.

Commit:
[ATIX] Add DebProfile to Profilemanager

Commit:
[ATIX] Move repofile to rhsm subpackage

Commit:
[ATIX] Generate enabled_repos from repofile

m-bucher committed:
[ATIX] Fix report enabled deb-repositories only

sbernhard committed:
[ATIX] Allow syspurpose file to be written

sbernhard committed:
[ATIX] This import is really necessary because it will be imported in rhsm/profile.py

hstct committed:
[ATIX] Add handling of params in repository urls

goarsna committed:
[ATIX] Remove failing method annotations

goarsna committed:
[ATIX] Use repofile from rhsm subpackage

goarsna committed:
[ATIX] Adjust katello script for Python3

quba42 committed:
[ATIX] Fix APT component selection

sbernhard committed:
[ATIX] Skip creating the yum.repos.d structure on debian based systems

(cherry picked from commit 9552aaac606629920df89f6536e4c0786ddef5dc)

m-bucher committed:
[ATIX] Fix failing profile upload with SCA enabled

Fixes OR-4461

Co-Authored-by: Bernhard Suttner <suttner@atix.de>

simmon-nplob committed:
Translated using Weblate (Korean)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (Japanese)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (Korean)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (Korean)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (Korean)

Currently translated at 100.0% (692 of 692 strings)

Co-authored-by: 김인수 <simmon@nplob.com>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/ja/
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/ko/
Translation: subscription-manager/subscription-manager-main

FAlexei committed:
Translated using Weblate (Russian)

Currently translated at 61.7% (427 of 692 strings)

Co-authored-by: Aleksey Fedorov <aleksejfedorov963@gmail.com>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/ru/
Translation: subscription-manager/subscription-manager-main

m-bucher committed:
[ATIX] Add deb repo-setting Signed-By

Fixes OR-4769

jirihnidek committed:
feat: Removed attach command and CLI option related to attach

* Card ID: CCT-425
* Removed attach command
* Removed --auto-attach and --autosubscribe CLI options of
  register command
* Removed --servicelevel CLI option of register command too,
  because this CLI option was possible to use only with
  --auto-attach CLI option. Thus, it does not make sense
  to keep this option
* Removed unit tests for attach command
* Modified unit tests for register command. Only SCA mode
  is considered.

jirihnidek committed:
feat: Removed D-Bus methods related to attach

* Card ID: CCT-425
* Removed interface: com.redhat.RHSM1.Attach
* Removed AutoAttach and PoolAttach
* Modified Register method
* Removed unit tests related to removed methods
* Modified unit tests of register D-Bus methods

jirihnidek committed:
feat: Removed attach service

* Card ID: CCT-425
* Removed attach service
* Removed unit tests related to attach service

jirihnidek committed:
feat: Remove references on auto-attach in man page

* Card ID: CCT-425
* Removed relevant references on auto-attach in man pages
  of subscription-manager
* Note: references on auto-atach (autoheal) was not removed,
  because it is different command and it only enable/disable
  autoheal option of rhsmcertd

jirihnidek committed:
feat: Remove attach from bash completion script

* Card ID: CCT-425
* Attach command was removed from bash complition script as well
  --auto-attach CLI option and --autosubscribe from register
  command

m-horky committed:
feat: Remove 'addons' subcommand(s)

* Card ID: CCT-592

chambridge committed:
Collect Azure VM Name and Resource Group Name as a cloud fact.

* Obtain the VM Name from the instance metadata API
* Obtain the Resource Group Name from the instance metadata API
* These items will improve deduplication logic within Subscription Watch metering

jirihnidek committed:
feat: Add rhsm group during installation of subman RPM

* Card ID: CCT-652
* rhsm group is created during installation of
  subscription-manager rpm
* System user conf file is installed to /usr/lib/sysuser.d
  directory.
* Existing consumer certs and keys are updated during
  installation process of rpm. It means that group owner
  of all pem files in /etc/pki/consumer is changed to rhsm
  and group can read these files. It is important for
  systems that are already registered.

m-horky committed:
feat: Disable anonymous cloud registration temporarily

* Card ID: CCT-654

Since the feature is not completed yet, we're disabling it client-side.

chambridge committed:
code review comments fixes
- update metadata and test.

jirihnidek committed:
feat: Create consumer cert & key owner by rhsm group

* Card ID: CCT-652
* When consumer certificate or key is created, then try to use
  rhsm group as a owner. When this group does not exist, then
  0 guid is used
* Modified also function checking and correcting permissions
  to consumer cert & key
* Not added any unit test for this case, because it would require
  to run unit tests as a root user

jirihnidek committed:
Merge pull request #3433 from chambridge/azure-deduplication-facts

Collect Azure VM Name and Resource Group Name as a cloud fact.

sbernhard committed:
Fixed spec file to list packages twice

m-bucher commented on src/rhsm/profile.py:
```diff
@@ -196,111 +196,28 @@ def collect(self) -> List[dict]:
-        self._enabled_repos: EnabledRepos = EnabledRepos(repo_file)
+        self._content = []
+        for repo_file_cls, _ in get_repo_file_classes():
+            repo_file = repo_file_cls()
```
Should this not rather be:
`repo_file = repo_file_cls(repo_file)`
This might have further implications, since apparently for now this expects to always use the `repo_file_cls()`'s default-value for `path`

jirihnidek committed:
Merge pull request #3438 from candlepin/mhorky/cct-654-disable-autoreg/el10

feat: Disable anonymous cloud registration temporarily

jvlcek committed:
docs: Change reverse proxy to proxy in man page

Change "reverse proxy" in rhsm.conf man page to simply "proxy"

Fixes: https://issues.redhat.com/browse/RHEL-52660
Fixes: https://issues.redhat.com/browse/CCT-682

jvlcek committed:
feat: Eliminate command 'remove' from subscription-manager

This PR eliminates the subscription-manager remove sub-command.

fixes: https://issues.redhat.com/browse/CCT-426

Glutexo committed:
doc: Update install and testing guide

The installation instructions now support RHEL in addition to Fedora. The guide has been tested on RHEL 9.2.

There are small touches to the commands in INSTALL.md too:

* PYTHONPATH is not a shell command, it doesn’t require explicit ./ for the current directory.
* An alias pointing to the python executable in the virtual environment makes it work without having to activate it.

The testing guide explicitly mentions requirements in the test-requirements.txt file and discourages from using ./setup.py test.

Card IDs:

* CCT-655
* CCT-656

m-horky committed:
chore: Format register.py

m-horky committed:
feat: Require SCA for registration

* Card ID: CCT-612

jirihnidek committed:
Merge pull request #3430 from candlepin/mhorky/cct-612/sca-only-registration

feat: Require SCA for registration

jirihnidek committed:
feat: Removed command "redeem" from subscription-manager

* Card ID: CCT-242
* Original change done by Martin Grünwald
* Removed command "redeem"
  * Removed all references from man page
  * Removed from bash completion script
  * Removed related unit tests

ptoscano committed:
Automatic commit of package [subscription-manager] release [1.30.0-1].

Created by command:

/usr/bin/tito tag --use-version=1.30.0

DuckBoss committed:
docs: remove references to removed commands

This patch removes remaining references to commands that were deprecated and removed.

 * Removed references to 'redeem' command
 * Removed references to 'attach' command
 * Fixed numbering for list of supported commands
 * Rephrased deprecated command section message
 * Update activation key messaging
 * Removed deprecated commands list
 * Adjusted wording slightly around attaching/auto-attaching subscriptions
 * Removed 'subscribe' from deprecated commands
 * Removed references to '--auto-attach/--servicelevel' register options

Card ID: CCT-545

ptoscano committed:
feat: forcefully switch automatic cloud registration to v1

- call the cloud endpoint without the new version, and with the Accept
  for the existing endpoint
- use cache.CloudTokenCache._get_from_server() directly: the v1
  autoregistration does not need any caching of the received token, so
  use the internal API of CloudTokenCache only to fetch the token
- the received token is the JWT token to use directly, without any
  inspection or unpacking: pass it directly to _auto_register_standard()
- drop the leftover commented code for autoregistration v2 for now

ptoscano committed:
Automatic commit of package [subscription-manager] release [1.30.1-1].

Created by command:

/usr/bin/tito tag

jvlcek committed:
Update the correct man page file.

m-horky committed:
test(ci): Improve container pre-test script

C10S, being a development build, doesn't always have all packages set up
correctly. As of now, two packages (rpm-devel, python3-rpm) depend on
different releases of rpm-libs, making the dnf command fail.

To make these issues more obvious next time, `set -euo pipefail` will
ensure the script fails instead of continuing.

jirihnidek committed:
feat: Remove auto-attach command

* Card ID: CCT-717
* Removed command that enabled/disabled autoheal option of
  consumer object on server side. The autoheal is useless
  in SCA mode and it will be removed later too.
* Updated creation of UEPConnection class and related
  stub class for unit testing
* Updated man page
* Updated bash completion script

quba42 committed:
[ATIX] Handle quoted baseurl

Candlepin 4.4 has started to deliver quote baseurl's. Apparently this
is considered a bug, but there is no telling when it will be fixed on
the candlepin side. In addition unquote() should not harm strings that
are not quoted, so this works either way.

(cherry picked from commit 805d69b459c8fa17939b7e94e3f55d75058dc3b8)

jirihnidek committed:
feat: dnf plugin - outsource uploading of profile to rhsmcertd.

* Card ID: CCT-505, RHEL-35656
* When rhsmcertd is running, then uploading of profile is
  outsourced to rhsmcertd process simply by sending SIGUSR1
  signal and dnf is not blocked by this activity
* When it is not possible to send SIGUSR1 signal, then
  try to send DNF profile in blocking mode
* This should also fix issue described in RHEL-35656

jvlcek committed:
feat: Remove autoheal functionality from rhsmcertd

Card ID: CCT-723

Removed autoheal sub-command from rhsmcertd because
autoheal is useless in pure SCA mode..

NorwayFun committed:
Translated using Weblate (Georgian)

Currently translated at 29.0% (201 of 692 strings)

Co-authored-by: Temuri Doghonadze <temuri.doghonadze@gmail.com>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/ka/
Translation: subscription-manager/subscription-manager-main

m-horky committed:
feat: Better automatic registration logging

m-horky committed:
fix: Handle Retry-After headers better for 429 responses

* Card ID: CCT-759

We have to ensure we normalize the headers before we search for the
Retry-After header.

rverdile committed:
feat: support registering specifying environments with activation keys

* Card-ID: CCT-762
* Removes validation to prevent specifying environment during
  registration with activation key

ptoscano committed:
Automatic commit of package [subscription-manager] release [1.30.2-1].

Created by command:

/usr/bin/tito tag

ptoscano committed:
chore: move autoreg waiting code in own function

Move the code that performs the waiting/delay during automatic
registration in its own function; this way it can be reused also in
other places.

This is only a code recfatoring, there is no behaviour changes.

ptoscano committed:
fix: perform autoreg waiting when performing standard autoreg

The initial delay when performing the standard autoregistration used to
be done directly in rhsmcertd (the C daemon). When the anonymous
autoregistration was implemented, since it has to start immediately,
that delay in rhsmcertd was dropped in favour of doing it "inline"
during the autoregistration. The waiting was added only during the
anonymous autoregistration flow and not during the standard
autoregistration.

Hence, add the autoregistration wait/delay before performing the
standard autoregistration, restoring the previous behaviour.

jirihnidek committed:
feat: Remove useless CLI options from list command

* Card ID: CCT-734
* Removed --available and --consumed CLI options and other options
  related to listing available and consumed subscriptions
* Only --installed (default) and --matches remains
* Modified man page
* Modified bash completion scripts
* Removed many unit tests and modified existing unit tests
  related to installed products. Unit tests consider only
  SCA mode

rverdile committed:
feat: add GetEnvironments method to DBus register

    * Card-ID: CCT-764
    * Method lists environments of the given org and returns a subset of the
      fields from candlepin
    * Provides a way to get the names of environments before registering
      in support of content templates

jirihnidek committed:
refactor: Moved some definition of lists from list.py

* Card ID: CCT-734
* cli_command/list.py contained several definitions of lists
  not related to list command. These definitions were moved
  to right Python modules. One list was renamed, because it
  is used only for testing ATM.

jirihnidek committed:
fix: Removed show_autosubscribe_output() method

* Card ID: CCT-734
* Removed show_autosubscribe_output() and usage of this
  method, because it does not make sense to use it in SCA mode

jirihnidek committed:
Merge pull request #3464 from candlepin/ptoscano/retry-after

fix: Handle Retry-After headers better for 429 responses

Commit:
feat/cct-874: Autocompletion for option --servicelevel removed
Autocompletion for option --servicelevel (sub-man register --servicelevel)
was removed.

CARD ID: CCT-874

Commit:
feat/cct-875: Options -i and --cert-interval were removed from rhsmcertd command autocompletion
Options -i and --cert-interval were removed from rhsmcertd command autocompletion.

CARD ID: CCT-875

jirihnidek committed:
feat: Added basic configuration for Packit

* Card ID: CCT-740
* Make builds on all fedoras and Centos Stream 10 ATM

DuckBoss committed:
feat: remove content access mode cache

This patch aims to remove references to the
'ContentAccessModeCache' and refactor/simplify related
code that checks this cache since entitlement is being phased
out and SCA will be the only content access mode.

Resolves: CCT-619

Signed-off-by: Jason Jerome <jajerome@redhat.com>

ptoscano committed:
fix: drop "autoattachinterval" from the rhsmcertd defaults

That configuration option was recently removed, so there is no more need
to specify a default value for it.

Fixes CCT-966

Followup of commit c124784a20276f1877248f15ed706050b7400dd5.

pkoprda committed:
Merge pull request #3472 from candlepin/ptoscano/CCT-966

fix: drop "autoattachinterval" from the rhsmcertd defaults

jirihnidek committed:
fix: Renamed integration-tests to cockpit-tests

* Card ID: CCT-741
* The new name describes better the purpose of the directory
* We are going to add real integration tests and there would
  be interference in names

jirihnidek committed:
feat: Add initial support for tmt

* Card ID: CCT-741
* Added basic configuration for tmt
  - subscription-manager is installed from COPR build
  - Python virtual environment is created
  - Install testing framework
  - Some simple smoke test is run (busctl calls RHSM D-Bus method)

Glutexo committed:
chore: Remove artifacts of attach

Removed what remained after removing the `attach` command. A system is now attached upon registration.

Card IDs:
* CCT-603

Signed-off-by: Štěpán Tomsa <stomsa@redhat.com>

grunwmar committed:
Feat CCT-965: Include timezone in the logs
In `subscription-manager/src/rhsm/logutil.py`i:

* New loggin formater class `RhsmISO8601Formatter` was made
to ensure that time stamp entry is in ISO-8810 format and
that it contains info about milliseconds and time zone.

CARD CCT-965

jstavel committed:
Card ID: CCT-731 - integration tests for DBus Register method

jirihnidek committed:
Merge pull request #3462 from rverdile/764

feat: add GetEnvironments method to DBus register

Glutexo committed:
chore: Remove artifacts of --auto-attach

Removed what remained after removing the `--auto-attach` option of `register`. A system is now always attached upon registration.

Card IDs:
* CCT-603

Signed-off-by: Štěpán Tomsa <stomsa@redhat.com>

Glutexo committed:
chore: Remove artifacts of autoheal

Removed what remained after removing the `autoheal` option from Action clients (ENT-5549).

Card IDs:
* CCT-603

Signed-off-by: Štěpán Tomsa <stomsa@redhat.com>

Glutexo committed:
chore: Remove artifacts of redeem

Removed what remained after removing the `redeem` command.

Card IDs:
* CCT-603

Signed-off-by: Štěpán Tomsa <stomsa@redhat.com>

Glutexo committed:
chore: Remove artifacts of import

Removed what remained after removing the `import` command.

Card IDs:
* CCT-603

Signed-off-by: Štěpán Tomsa <stomsa@redhat.com>

Glutexo committed:
chore: Remove artifacts of remove

Removed what remained after removing the `remove` command.

Card IDs:
* CCT-603

Signed-off-by: Štěpán Tomsa <stomsa@redhat.com

jirihnidek committed:
feat: Enable register with environment names and environment types

* Allow to register also with environment names. It is possible
  to use environment names with username & password and
  activation-keys & organization authentication
* It is not allowed to use `environmets` and `environment_names`
  registration options together, because it is not possible to use
  env. IDs and names together on candlepin server
* Introduce environment_type as another registration_option
  * When consumer object is returned and it contains environments,
    then all environments are checked if type of environment
    matches given environment_type. If type is missing or is
    different, then system is unregistered and exception is
    raised
* Modified few unit tests related to environments
* Added few unit tests

pkoprda committed:
chore: Remove the --token authentication

* Card ID: CCT-1038

The token-based authentication method was deprecated in RHEL 9.2 and
RHEL 8.8. As this feature is no longer relevant, and due to its
deprecation, it is safe to remove `--token` starting with RHEL 10.

m-horky committed:
chore: Remove auto-assign CI job

The workflow has had lesser and lesser impact on the ways our team picks tasks
waiting for review. We are beyond the point of diminishing returns, it's
easier to drop it completely and rely on other means.

jirihnidek committed:
fix: Fixed integration tests

* Card ID: CCT-741
* .fmf/version file was in wrong subdirectory (./systemtest)
* test.sh did not have execute bit

jirihnidek committed:
Merge pull request #3482 from candlepin/mhorky/no-auto-assign

chore: Remove auto-assign CI job

jirihnidek committed:
fix: Added missing python packages

jirihnidek committed:
Merge pull request #3477 from candlepin/jstavel/dbus-integration-tests

Integration Tests for DBus Register Method

jirihnidek committed:
Automatic commit of package [subscription-manager] release [1.30.3-1].

Created by command:

/usr/bin/tito tag

jirihnidek committed:
Merge pull request #3480 from candlepin/jhnidek/reg_env_names

feat: Enable register with environment names & type

sbernhard committed:
[ATIX] Fix test_repolib tests because of apt/deb822

sbernhard committed:
[ATIX] Fix GPG trusted repolib test

Atalanttore committed:
Translated using Weblate (German)

Currently translated at 76.4% (529 of 692 strings)

Co-authored-by: Ettore Atalan <atalanttore@googlemail.com>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/de/
Translation: subscription-manager/subscription-manager-main

simmon-nplob committed:
Translated using Weblate (Korean)

Currently translated at 100.0% (692 of 692 strings)

Co-authored-by: 김인수 <simmon@nplob.com>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/ko/
Translation: subscription-manager/subscription-manager-main

salvatore-cocuzza committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (692 of 692 strings)

Co-authored-by: Salvatore Cocuzza <info@salvatorecocuzza.it>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/it/
Translation: subscription-manager/subscription-manager-main

HarmfulBreeze committed:
Translated using Weblate (French)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (French)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (French)

Currently translated at 100.0% (692 of 692 strings)

Translated using Weblate (French)

Currently translated at 100.0% (692 of 692 strings)

Co-authored-by: Léane GRASSER <leane.grasser@proton.me>
Translate-URL: https://translate.fedoraproject.org/projects/subscription-manager/subscription-manager-main/fr/
Translation: subscription-manager/subscription-manager-main

ptoscano committed:
New extraction for translatable strings

ptoscano committed:
ci: use official c10s container image

Now that CentOS Stream 10 is officially released, switch to it from the
old development version of it.

sbernhard committed:
[ATIX] Remove package-profile-upload while building RPM

The script is only required in debian/ubuntu installations

sbernhard committed:
WIP: temp. disable centos10 because of failed build

m-bucher reviewed (approved):
If it works, I guess this is OK (even without detailed review :sweat_smile: )

sbernhard commented:
root@debian12:~# subscription-manager register --org="Example" --name="debian12.master.dev.atix" --activationkey="ac_Debian12" --force
Unregistering from: or.master.dev.atix:443/rhsm
The system with UUID f42fbb80-c766-4e3e-b002-61473fe77738 has been unregistered
All local data removed
The system has been registered with ID: b3e6456a-d8c8-490a-b557-1aa5e913ac64
The registered system name is: debian12.master.dev.atix
